### PR TITLE
fix: revision styles for workflow

### PR DESCRIPTION
**Pull request by lawvs**
# Description

This pull request involves fixing some minor style issues.

- Fix incorrect model position

https://github.com/langgenius/dify/assets/18554747/130a2843-6991-4e49-a493-09c4d0835592

- Fix incorrect cursor style in the Panel component

https://github.com/langgenius/dify/assets/18554747/48db1deb-af44-49e0-a2f5-2dd313148253

- Support toggle the Feature panel

Screenshot 2024-05-05 at 03 06 01

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement

# How Has This Been Tested?

- [ ] See video above

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

File: web/app/components/workflow/nodes/_base/components/memory-config.tsx [hunk 1]
```diff
@@ -50,7 +50,10 @@ type Props = {
   canSetRoleName?: boolean
 }
 
-const MEMORY_DEFAULT: Memory = { window: { enabled: false, size: WINDOW_SIZE_DEFAULT } }
+const MEMORY_DEFAULT: Memory = {
+  window: { enabled: false, size: WINDOW_SIZE_DEFAULT },
+  query_prompt_template: '',
+}
```
Comment (lawvs) [inline web/app/components/workflow/nodes/_base/components/memory-config.tsx:56]:
```
Property 'query_prompt_template' is missing in type '{ window: { enabled: false; size: number; }; }' but required in type 'Memory'.ts(2741)
```

It is recommended to add type check to the CI.

File: web/.vscode/settings.example.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "prettier.enable": false,
   "editor.formatOnSave": true,
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
+    "source.fixAll.eslint": "explicit"
```
Comment (lawvs) [inline web/.vscode/settings.example.json:5]: > Code Actions' behavior on save in the editor has been modified from Booleans to an enum with the following options:
`always`: Code Actions with always be run on any kind of save (autosave when exiting or changing focus, explicit save, etc).
`explicit`: Code Actions will be run only when explicitly saving.
`never`: Code Actions will never be run on save.

See https://github.com/microsoft/vscode/issues/163920#issuecomment-1721561959

File: web/app/components/header/account-setting/model-provider-page/model-parameter-modal/index.tsx [hunk 3]
```diff
@@ -152,7 +152,7 @@ const ModelParameterModal: FC<ModelParameterModalProps> = ({
     <PortalToFollowElem
       open={open}
       onOpenChange={setOpen}
-      placement='bottom-end'
+      placement='left'
```
Comment (lawvs) [inline web/app/components/header/account-setting/model-provider-page/model-parameter-modal/index.tsx:155]: The model parameter now properly stays on the left.

https://github.com/langgenius/dify/assets/18554747/73fe4d36-cd82-4cb5-bbad-f6c7ea5de828

File: web/app/components/workflow/header/index.tsx [hunk 4]
```diff
@@ -58,13 +58,13 @@ const Header: FC = () => {
...
     const {
+      showFeaturesPanel,
       isRestoring,
       setShowFeaturesPanel,
     } = workflowStore.getState()
     if (getNodesReadOnly() && !isRestoring)
       return
-
-    setShowFeaturesPanel(true)
+    setShowFeaturesPanel(!showFeaturesPanel)
```
Comment (lawvs) [inline web/app/components/workflow/header/index.tsx:97]: The features panel can now be closed by clicking on it again.


https://github.com/langgenius/dify/assets/18554747/7e275461-25e4-4866-a77d-5ebdaee00609

File: web/app/components/workflow/nodes/_base/components/field.tsx [hunk 5]
```diff
@@ -27,10 +27,10 @@ const Filed: FC<Props> = ({
     toggle: toggleFold,
   }] = useBoolean(true)
   return (
-    <div className={cn(inline && 'flex justify-between items-center', supportFold && 'cursor-pointer')}>
+    <div className={cn(inline && 'flex justify-between items-center')}>
       <div
         onClick={() => supportFold && toggleFold()}
-        className='flex justify-between items-center'>
+        className={cn('flex justify-between items-center', supportFold && 'cursor-pointer')}>
```
Comment (lawvs) [inline web/app/components/workflow/nodes/_base/components/field.tsx:37]: The `supportFold` panel now can display cursor style correctly.

https://github.com/langgenius/dify/assets/18554747/3ad01c8f-cb6f-4343-9e7b-4d912c033326

**Comment by crazywoola** [comment]
@lawvs Hello please fix the conflict, I will review it asap.

**Comment by lawvs** [comment]
@crazywoola Done

**Review by crazywoola — APPROVED** [review]

**Pull request merged**